src/DynamoCoreWpf/ViewModels/Core/Converters/SerializationConverters.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Linq;
+using Dynamo.Graph.Annotations;
+using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Notes;
+using Dynamo.Nodes;
+using Dynamo.Utilities;
+using Dynamo.ViewModels;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Type = System.Type;
+
+namespace Dynamo.Wpf.ViewModels.Core.Converters
+{
+    /// <summary>
+    /// The AnnotationConverter is used to serialize and deserialize AnnotationModels.
+    /// The SelectedModels property on AnnotationModel is a list of references
+    /// to ModelBase objects. During serialization we want to refer to these objects
+    /// by their ids. During deserialization, we use the ReferenceResolver to
+    /// find the correct ModelBase instances to reference.
+    /// </summary>
+    public class AnnotationViewModelConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(AnnotationViewModel);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var obj = JObject.Load(reader);
+            var title = obj["Title"].Value<string>();
+            //if the id is not a guid, makes a guid based on the id of the model
+            Guid annotationId = GuidUtility.tryParseOrCreateGuid(obj["Id"].Value<string>());
+
+            // This is a collection of string Guids, which
+            // should be accessible in the ReferenceResolver.
+            var models = obj["Nodes"].Values<JValue>();
+
+            var existing = models.Select(m =>
+            {
+                Guid modelId = GuidUtility.tryParseOrCreateGuid(m.Value<string>());
+
+                return serializer.ReferenceResolver.ResolveReference(serializer.Context, modelId.ToString());
+            });
+
+            var nodes = existing.Where(m => typeof(NodeModel).IsAssignableFrom(m.GetType())).Cast<NodeModel>();
+            var notes = existing.Where(m => typeof(NoteModel).IsAssignableFrom(m.GetType())).Cast<NoteModel>();
+
+            var anno = new AnnotationModel(nodes, notes);
```

Comment:
I am not sure how this works in deserilization. I think it should be straight forward.